app/routes.py

```diff
@@ -1,2 +1,233 @@
-from flask import Blueprint
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+from app import db
+from datetime import datetime
+from sqlalchemy import asc, desc
+from app import slack_app
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+
+    request_body = request.get_json()
+
+    if ("title" not in request_body.keys() or
+        "description" not in request_body.keys() or
+        "completed_at" not in request_body.keys()):
+        return {"details": "Invalid data"}, 400
+
+    else:
+        new_task = Task(title = request_body["title"],
+                        description = request_body["description"],
+                        completed_at = request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {
+            "task": new_task.to_json()
+            }, 201
+
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def read_by_title():
+
+    query_string = request.args.get("sort")
+
+    if query_string == "asc":
+        tasks = Task.query.order_by(asc(Task.title))
+    
+    elif query_string == "desc":
+        tasks = Task.query.order_by(desc(Task.title))
+
+    else:
+        tasks = Task.query.all()
+    
+    task_response = []
+
+    for task in tasks:
+        task_response.append(task.to_json())
+    
+    return jsonify(task_response), 200
+
+
+@tasks_bp.route("/<int:task_id>", methods=["GET"], strict_slashes=False)
+def read_by_id(task_id):
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return("", 404)
```

Comment:
Nice!! You can also reduce these lines of code by using Flask's `get_or_404()` method
```suggestion
    task = Task.query.get_or_404(task_id)

```

Reply:
This method can be used pretty much anywhere you're grabbing an id.